### PR TITLE
chore(compiler): Add `fromNumber(<literal>)` warning for `short` types

### PR DESCRIPTION
**Pull request by spotandjake**
This pr makes the `fromNumber(<literal>)` warning work on all types.

On a side note I think we broke the implementation of this check when we added default args and this pr also fixes that issue.

I merged the `FromNumberLiteralXXX` warnings into a single warning. I don't know if we want to make this a breaking change as the warning numbers have changed (I feel like we should consider it as one). 

closes: #2019

### CODE REVIEW

File: compiler/src/typed/typed_well_formedness.re [hunk 1]
```diff
@@ -302,8 +302,12 @@ module WellFormednessArg: TypedtreeIter.IteratorArgument = {
           ],
         )
           when
-            modname == "Int32"
+            modname == "Int8"
```
Comment (alex-snezhko) [inline compiler/src/typed/typed_well_formedness.re:305]: Should we apply these to `BigInt` and `Rational` as well?

Comment (spotandjake) [reply compiler/src/typed/typed_well_formedness.re:305]: Yeah in that case I'm going to make an `isNumber` function

Comment (spotandjake) [reply compiler/src/typed/typed_well_formedness.re:309]: Made that change

**Review by ospencer — COMMENTED** [review]
Can you add tests for the short nums?

File: compiler/src/typed/typed_well_formedness.re [hunk 2]
```diff
@@ -289,42 +289,62 @@ module WellFormednessArg: TypedtreeIter.IteratorArgument = {
...
+            || modname == "BigInt" =>
         // NOTE: Due to type-checking, we shouldn't need to worry about ending up with a FloatXX value and a Const_number_float
         let n_str =
           switch (n) {
           | Const_number_int(nint) => Int64.to_string(nint)
           | Const_number_float(nfloat) => Float.to_string(nfloat)
-          | _ => failwith("Impossible")
+          | Const_number_rational({rational_num_rep, rational_den_rep}) =>
+            Printf.sprintf("%s/%s", rational_num_rep, rational_den_rep)
+          | Const_number_bigint({bigint_rep}) => bigint_rep
```
Comment (phated) [inline compiler/src/typed/typed_well_formedness.re:328]: Is this exhaustive? If so, you can streamline the first pattern match

**Review by phated — CHANGES_REQUESTED** [review]
One thing

File: compiler/src/typed/typed_well_formedness.re (hunk 2 above)
Comment (spotandjake) [reply compiler/src/typed/typed_well_formedness.re:328]: What do you mean by that?

Comment (phated) [reply compiler/src/typed/typed_well_formedness.re:328]: On lines 297-301 you are pattern matching and then assigning with `as n` but if this is exhaustive, those can just use `n` in the match. I'll make a suggestion above

File: compiler/src/typed/typed_well_formedness.re [hunk 3]
```diff
@@ -289,42 +289,62 @@ module WellFormednessArg: TypedtreeIter.IteratorArgument = {
...
               {
                 exp_desc:
                   TExpConstant(
                     Const_number(
-                      (Const_number_int(_) | Const_number_float(_)) as n,
+                      (
+                        Const_number_int(_) | Const_number_float(_) |
+                        Const_number_rational(_) |
+                        Const_number_bigint(_)
+                      ) as n,
```
Comment (phated) [inline compiler/src/typed/typed_well_formedness.re:301]:
```suggestion
                      n
```
Suggestion for discussion below

**Review by ospencer — APPROVED** [review]

**Review by phated — APPROVED** [review]

**Pull request merged**